website/integrations: improve grafana docs (#14408)
goauthentik/authentik: website/integrations/services/grafana/index.mdx

    GF_AUTH_OAUTH_AUTO_LOGIN: "true"
    # Optionally map user groups to Grafana roles
    GF_AUTH_GENERIC_OAUTH_ROLE_ATTRIBUTE_PATH: "contains(groups, 'Grafana Admins') && 'Admin' || contains(groups, 'Grafana Editors') && 'Editor' || 'Viewer'"

>     # If Grafana is running behind a reverse proxy, you will also need (https://github.com/goauthentik/authentik/issues/8673):

Reviewer: ```suggestion
    # Required if Grafana is running behind a reverse proxy
```

### Grafana Configuration Considerations

> Make sure in your configuration that `root_url` is set correctly (as shown in the Docker config above), otherwise your redirect url might get processed incorrectly. For example, if your grafana instance is running on the default configuration and is accessible behind a reverse proxy at `https://grafana.company`, your redirect url will end up looking like this, `https://grafana.company/login/generic_oauth`.

Reviewer: ```suggestion
To ensure redirects work correctly in Grafana, make sure the `root_url` in your configuration accurately reflects how users access Grafana through your reverse proxy. For example, if your Grafana instance is behind a proxy and accessed at `https://grafana.company`, set `root_url` to `https://grafana.company`. This ensures that OAuth and other redirects use the correct URL, such as `https://grafana.company/login/generic_oauth`, instead of defaulting to something like `localhost:3000`.
```
